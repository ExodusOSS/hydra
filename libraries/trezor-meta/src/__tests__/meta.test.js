const meta = require('../meta')

const mockAssets = {
  bitcoin: { assetType: 'BITCOIN_LIKE' },
  ethereum: { assetType: 'ETHEREUM_LIKE' },
  token1: { assetType: 'ETHEREUM_ERC20' },
  token2: { assetType: 'ETHEREUM_ERC20' },
}

describe('Trezor meta', () => {
  test('should match snapshot', () => {
    const {
      isSegwit,
      isBip84,
      isSupportedByTrezor,
      isSupportedOnlyByModelT,
      isSupportedFirmware,
      ...staticProps
    } = meta(mockAssets)
    expect(staticProps).toMatchSnapshot()
  })
})
