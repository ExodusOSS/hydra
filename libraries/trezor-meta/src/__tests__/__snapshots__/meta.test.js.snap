exports[`Trezor meta should match snapshot 1`] = `
{
  "ASSETS_BY_MODEL": {
    "1": [
      "bcash",
      "bitcoin",
      "dogecoin",
      "ethereum",
      "ethereumclassic",
      "litecoin",
      "stellar",
      "token1",
      "token2",
      "zcash",
      "dash",
      "decred",
      "digibyte",
    ],
    "Safe 3": [
      "bcash",
      "bitcoin",
      "cardano",
      "dogecoin",
      "ethereum",
      "ethereumclassic",
      "litecoin",
      "monero",
      "ripple",
      "stellar",
      "token1",
      "token2",
      "zcash",
    ],
    "Safe 5": [
      "bcash",
      "bitcoin",
      "cardano",
      "dogecoin",
      "ethereum",
      "ethereumclassic",
      "litecoin",
      "monero",
      "ripple",
      "stellar",
      "token1",
      "token2",
      "zcash",
    ],
    "T": [
      "bcash",
      "bitcoin",
      "cardano",
      "dash",
      "decred",
      "digibyte",
      "dogecoin",
      "ethereum",
      "ethereumclassic",
      "litecoin",
      "monero",
      "ripple",
      "stellar",
      "token1",
      "token2",
      "zcash",
    ],
  },
  "BASE_SUPPORTED_ASSETS": [
    "bcash",
    "bitcoin",
    "dogecoin",
    "ethereum",
    "ethereumclassic",
    "litecoin",
    "stellar",
    "token1",
    "token2",
    "zcash",
  ],
  "FIRMWARE_ASSETS_BY_MODEL": {
    "1": [
      "bitcoin",
      "bcash",
      "dash",
      "dogecoin",
      "digibyte",
      "decred",
      "litecoin",
      "vertcoin",
      "zcash",
      "dogecoin",
      "vertcoin",
      "ravencoin",
      "ethereum",
      "ethereumclassic",
      "lisk",
      "stellar",
      "nem",
      "token1",
      "token2",
    ],
    "Safe 3": [
      "bitcoin",
      "bcash",
      "dogecoin",
      "monero",
      "litecoin",
      "zcash",
      "dogecoin",
      "ravencoin",
      "ethereum",
      "ethereumclassic",
      "stellar",
      "ripple",
      "tezos",
      "cardano",
      "token1",
      "token2",
    ],
    "Safe 5": [
      "bitcoin",
      "bcash",
      "dogecoin",
      "monero",
      "litecoin",
      "zcash",
      "dogecoin",
      "ravencoin",
      "ethereum",
      "ethereumclassic",
      "stellar",
      "ripple",
      "tezos",
      "cardano",
      "token1",
      "token2",
    ],
    "T": [
      "bitcoin",
      "bcash",
      "dash",
      "dogecoin",
      "digibyte",
      "monero",
      "decred",
      "litecoin",
      "vertcoin",
      "zcash",
      "dogecoin",
      "vertcoin",
      "ravencoin",
      "ethereum",
      "ethereumclassic",
      "lisk",
      "stellar",
      "ripple",
      "tezos",
      "nem",
      "cardano",
      "token1",
      "token2",
    ],
  },
  "FUTURE_SUPPORTED_ASSETS": [
    "tezos",
    "ravencoin",
  ],
  "SUPPORTED_ASSETS": [
    "bcash",
    "bitcoin",
    "cardano",
    "dash",
    "decred",
    "digibyte",
    "dogecoin",
    "ethereum",
    "ethereumclassic",
    "litecoin",
    "monero",
    "ripple",
    "stellar",
    "token1",
    "token2",
    "zcash",
  ],
  "SUPPORTED_MODELS": [
    "Safe 5",
    "Safe 3",
    "T",
    "1",
  ],
  "SUPPORTED_ONLY_BY_MODEL_T": [
    "cardano",
    "monero",
    "ripple",
  ],
}
`;
